lib/airbrake/worker.ex

```diff
@@ -84,12 +84,24 @@ defmodule Airbrake.Worker do
 
   defp send_report(exception, stacktrace, options) do
     unless ignore?(exception) do
-      payload = Airbrake.Payload.new(exception, stacktrace, options)
+      enhanced_options = build_options(options)
+      payload = Airbrake.Payload.new(exception, stacktrace, enhanced_options)
       json_encoder = Application.get_env(:airbrake, :json_encoder, Poison)
       HTTPoison.post(notify_url(), json_encoder.encode!(payload), @request_headers)
     end
   end
 
+  defp build_options(current_options) do
+    case get_env(:options) do
+      base_options when is_function(base_options, 1) ->
+        base_options.(current_options)
+      shared_options when is_list(shared_options) ->
+        Keyword.merge(shared_options, current_options)
```

Comment:
It’d be nice to have a test for at least this line of code, making sure the `current_options` win out.  (They do, right?)  I’m really inclined to extract both of these functions (maybe into an `Airbrake.Report` module) so that they can be tested more easily.

But I’m fine with how it is now.  It’s simple enough that I’m sure it works; it’d just be nice to have the tests for regression.

Reply:
I can see the wish. My goal right now is to keep it as simple as possible in the hopes that the it will be accepted in the upstream project.